lib/matplotlib/backends/backend_qt.py

```diff
@@ -121,6 +120,10 @@ def _create_qApp():
             except AttributeError:  # Only for Qt>=5.14.
                 pass
             qApp = QtWidgets.QApplication(["matplotlib"])
+            if sys.platform == "darwin":
```

Comment:
Is there a harm in setting this regardless of platform? I just tested on a Linux box and it didn't seem to have an impact.

Ditto to the block below, where even if we are on mac, I don't think it hurts anything to call `setWindowIcon` on the windows.

Reply:
I don't have a problem removing this one.

With the other one, theoretically window icons on macOS are supposed to only be set along with a file path when the window represents a file. However, if you don't set the file path, the icon won't show. So while it's an unnecessary call that's semantically incorrect, it's effectively a noop.

So I guess it's really up to your priorities: semantics? Performance? Less code? Up to you!

Reply:
Here's the documentation from Qt re: `QWidget::setWindowIcon()`:

https://doc.qt.io/qt-5/qwidget.html#windowIcon-prop

![image](https://user-images.githubusercontent.com/458871/143986049-27296c78-7e65-467f-b2e5-0abe3e341b87.png)

Reply:
Does it make sense to keep Darwin for now, and if someone wants this on another system they can add it?